bids_to_msd_dataset_conversion.py

```diff
@@ -0,0 +1,290 @@
+parser.add_argument('--include_sessions', nargs='+', required=False, type=str, default=None,
+                        help='Sessions (in proper BIDS suffixes) to include in the dataset. Note that only these sessions will be picked to create the dataset. '
+                        'If not used then all sessions will be included. Example: --include_sessions ses-01 ses-02')
+# argument that accepts a list of contrasts to include in the dataset
+parser.add_argument('--include_contrasts', nargs='+', required=False, type=str, default=None,
+                        help='Contrasts (in proper BIDS suffixes) to include in the dataset. Note that only these sessions will be picked to create the dataset. '
+                        'If not used then all contrasts will be included. Example: --contrasts T1w T2w')
+
+args = parser.parse_args()
+
+root = args.path_data
+output_dir = args.path_out
+train_ratio, val_ratio, test_ratio = args.split
+
+# set the random number generator seed
+rng = np.random.default_rng(args.seed)
+
+# check if participants.tsv exists 
+if not os.path.exists(os.path.join(root, 'participants.tsv')):
+    raise FileNotFoundError("participants.tsv file not found in the dataset directory. Cannot proceed with subject selection.")
+
+# Get all subjects from participants.tsv
+subjects_df = pd.read_csv(os.path.join(root, 'participants.tsv'), sep='\t')
+subjects = subjects_df['participant_id'].values.tolist()
+logger.info(f"Total number of subjects in the dataset: {len(subjects)}")
+
+# NOTE: sklearn does not have function for training, validation and test split. 
+# So, we use the following workaround to get the required splits
+# Get only the training and test splits initially
+train_subjects, test_subjects = train_test_split(subjects, test_size=test_ratio, random_state=args.seed)
+# Use the training split to further split into training and validation splits
+train_subjects, val_subjects = train_test_split(train_subjects, test_size=val_ratio/(train_ratio + val_ratio), 
+                                                    random_state=args.seed,)
+
+
+# boiler plate keys to be defined in the dataset.json
+params = {}
+params["description"] = "My awesome task"       # TODO: Add the name of the data
+params["labels"] = {
+    "0": "background",
+    "1": "sc-lesion"        # TODO: Define the classes to be be segmented
+    }
+params["license"] = "xxx"
+params["modality"] = {
+    "0": "MRI"
+    }
+params["name"] = "sci-zurich"       # TODO: Add the name of the dataset
+params["reference"] = "N/A"
+params["tensorImageSize"] = "3D"
+
+subjects_dict = {
+    "training": train_subjects,
+    "validation": val_subjects,
+    "test": test_subjects
+}
+
+# loop through the training, validation and test splits and create dictionaries that contain the 
+# (image, label) pairs for each subject
+for name, subs_list in subjects_dict.items():
+
+    temp_list = []
+    for subject_no, subject in enumerate(tqdm(subs_list, desc='Loading Volumes')):
```

Comment:
Now, `'Loading Volumes'` is printed 3x times to CLI (for train/valid/test). At first look, it is not clear that the first progress bar is related to the training fraction etc. Maybe, `desc` could be modified:
```
desc=f'Loading {name} Volumes'
```

Reply:
Added in https://github.com/ivadomed/templates/pull/1/commits/bf260b414fbd197f56b793c394cf5b5b7108a65a!